### PR TITLE
[review] create package review services

### PR DESCRIPTION
**Pull request by scalabw**
https://trello.com/c/9sXQgCuQ/2849-services-mocks

**Detailed purpose of the PR**

This PR Create a new package called review-service. It will be used by the web and mobile to make API call to the backend.
It also contains the fixtures and the Mocked services

**Result and observation**

To avoid duplicating the services and mocks in the app and the mobile app a seperated package has been created.

### CODE REVIEW

File: packages/@coorpacademy-app-review/src/actions/api/fetch-correction.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import type {Dispatch} from 'redux';
 import buildTask from '@coorpacademy/redux-task';
 import get from 'lodash/fp/get';
+import type {CorrectionFromAPI} from '@coorpacademy/review-services/es/types/services-types';
+import {ThunkOptions} from 'src/types/common';
```
Comment (StefanoA1) [inline packages/@coorpacademy-app-review/src/actions/api/fetch-correction.ts:5]:
```suggestion
import type {ThunkOptions} from 'src/types/common';
```

**Review by StefanoA1 — CHANGES_REQUESTED** [review]

**Review by StefanoA1 — APPROVED** [review]

**Review by StefanoA1 — APPROVED** [review]

File: packages/@coorpacademy-review-services/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "allowJs": true,
+    "declaration": true,
+    "declarationMap": true,
+    "skipLibCheck": true,
+    "jsx": "preserve",
+    "allowSyntheticDefaultImports": true,
+    // https://stackoverflow.com/questions/39423054/react-jsx-file-giving-error-cannot-read-property-createelement-of-undefined#comment103952221_56015464
+    "esModuleInterop": true,
```
Comment (StefanoA1) [inline packages/@coorpacademy-review-services/tsconfig.json:13]: je pense que tu peux enlever ça (pas de tsx/jsx)

File: packages/@coorpacademy-review-services/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "compilerOptions": {
+    "target": "es2020",
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "allowJs": true,
+    "declaration": true,
+    "declarationMap": true,
+    "skipLibCheck": true,
+    "jsx": "preserve",
```
Comment (StefanoA1) [inline packages/@coorpacademy-review-services/tsconfig.json:10]: pareil

File: packages/@coorpacademy-review-services/tsconfig.json (hunk 2 above)
Comment (StefanoA1) [inline packages/@coorpacademy-review-services/tsconfig.json:13]: you can take this out (no tsx/jsx)
```suggestion
```

File: packages/@coorpacademy-app-review/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@coorpacademy/app-review",
-  "version": "0.9.7",
+  "version": "0.10.1-alpha.0",
```
Comment (chrisdugne) [inline packages/@coorpacademy-app-review/package.json:3]: attention, tu as du faire une `canary`, il ne faut pas commit cette version

File: packages/@coorpacademy-review-services/tsconfig.json (hunk 3 above)
Comment (StefanoA1) [inline packages/@coorpacademy-review-services/tsconfig.json:10]: same, no jsx
```suggestion
```

File: packages/@coorpacademy-review-services/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{
+  "compilerOptions": {
+    "target": "es2020",
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "allowJs": true,
```
Comment (StefanoA1) [inline packages/@coorpacademy-review-services/tsconfig.json:6]: no js
```suggestion
```

File: packages/@coorpacademy-review-services/tsconfig.lib.json [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "emitDeclarationOnly": true,
+    "outDir": "lib"
+  },
```
Comment (StefanoA1) [inline packages/@coorpacademy-review-services/tsconfig.lib.json:6]:
```suggestion
    "outDir": "lib",
    "incremental": true,
    "noErrorTruncation": true
  },
```

**Review by godu — CHANGES_REQUESTED** [review]

File: packages/@coorpacademy-review-services/package.json [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    "clean": "concurrently \"npm run clean:commonjs\" \"npm run clean:es\"",
+    "ava": "ava",
+    "test": "npm run lint && npm run test:unit",
+    "lint": "npm run eslint",
+    "test:unit": "nyc npm run ava",
+    "eslint": "eslint . --ext .js,.json,.ts,.tsx --cache --cache-location \"node_modules/.cache/.eslintcache\"",
+    "typecheck": "tsc --noEmit true"
+  },
+  "engines": {
+    "node": ">=12.22.12"
```
Comment (chrisdugne) [inline packages/@coorpacademy-review-services/package.json:34]:
```
    "node": ">=16.15.0"
```
= version de node des autres packages

File: packages/@coorpacademy-app-review/src/actions/api/test/fetch-slides-to-review-by-skill-ref.test.ts [hunk 8]
```diff
@@ -1,23 +1,23 @@
 import test from 'ava';
-import type {StoreState} from '../../../reducers';
-import type {Services} from '../../../types/common';
+import type {Services} from '@coorpacademy/review-services';
+import {
+  fetchSlidesToReviewBySkillRefResponse,
+  postAnswerResponses,
```
Comment (jomaora) [inline packages/@coorpacademy-app-review/src/actions/api/test/fetch-slides-to-review-by-skill-ref.test.ts:5]: il y a des objets qui sont dans services-mock et qui ne devrait plus l'être
postAnswerResponses (l:98) et incorrectFreeTextPostAnswerResponse (l:307) de l'index. Déplace le vers app-review. Merci

**Review by jomaora — CHANGES_REQUESTED** [review]

File: packages/@coorpacademy-app-review/src/actions/api/test/fetch-slides-to-review-by-skill-ref.test.ts (hunk 8 above)
Comment (scalabw) [reply packages/@coorpacademy-app-review/src/actions/api/test/fetch-slides-to-review-by-skill-ref.test.ts]: done !

File: packages/@coorpacademy-app-review/src/reducers/data/test/progression.test.ts [hunk 9]
```diff
@@ -1,12 +1,12 @@
 import test from 'ava';
+import {postAnswerResponses, postProgressionResponse} from '@coorpacademy/review-services-mocks';
```
Comment (jomaora) [inline packages/@coorpacademy-app-review/src/reducers/data/test/progression.test.ts:2]: il y a encore un `postAnswerResponses`

File: packages/@coorpacademy-app-review/src/test/index.test.tsx [hunk 10]
```diff
@@ -4,10 +4,10 @@ import test from 'ava';
 import type {ExecutionContext} from 'ava';
 import React from 'react';
 import {render, fireEvent, act} from '@testing-library/react';
+import {services} from '@coorpacademy/review-services-mocks';
+import {sleep} from '@coorpacademy/review-services/es/tools/sleep';
```
Comment (jomaora) [inline packages/@coorpacademy-app-review/src/test/index.test.tsx:8]: `sleep` should not be exported by the `review-services` lib (it's not a service). Please check if this function is used internally by the `review-services`. If not, the function must be deleted there and defined here (for internal use of `app-review`).

File: packages/@coorpacademy-app-review/sandbox/translation.mock.ts [hunk 11]
```diff
@@ -0,0 +1,4 @@
+import {mockTranslate} from '@coorpacademy/translate';
```
Comment (jomaora) [inline packages/@coorpacademy-app-review/sandbox/translation.mock.ts:1]: plase move this file to `src/tests/utils`. I'd rather see tests importing this mock from utils, than importing it from the sandbox enviroment.

File: packages/@coorpacademy-app-review/src/views/skills/test/skills.test.ts [hunk 12]
```diff
@@ -4,8 +4,8 @@ import map from 'lodash/fp/map';
 import omit from 'lodash/fp/omit';
 import {StoreState} from '../../../reducers';
 import {mapStateToSkillsProps} from '..';
-import {translate} from '../../../test/util/services.mock';
 import {skin} from '../../slides/test/fixtures/skin';
+import {translate} from '../../../../sandbox/translation.mock';
```
Comment (jomaora) [inline packages/@coorpacademy-app-review/src/views/skills/test/skills.test.ts:8]: to be imported from src/util/

File: packages/@coorpacademy-app-review/src/views/slides/test/button-revising.on-click.test.ts [hunk 13]
```diff
@@ -11,19 +13,14 @@ import {SLIDE_FETCH_REQUEST, SLIDE_FETCH_SUCCESS} from '../../../actions/api/fet
...
 import {StoreState} from '../../../reducers';
 import {mapStateToSlidesProps} from '..';
-import {
-  getChoicesCorrection,
-  postAnswerResponses,
-  services,
-  translate
-} from '../../../test/util/services.mock';
-import type {ProgressionFromAPI} from '../../../types/common';
+import {translate} from '../../../../sandbox/translation.mock';
```
Comment (jomaora) [inline packages/@coorpacademy-app-review/src/views/slides/test/button-revising.on-click.test.ts:16]: idem

File: packages/@coorpacademy-app-review/src/views/slides/test/on-quit-popin.on-click.test.ts [hunk 14]
```diff
@@ -1,18 +1,16 @@
...
 import {createTestStore} from '../../../actions/test/create-test-store';
 import {CLOSE_POPIN} from '../../../actions/ui/quit-popin';
-import {
-  incorrectFreeTextPostAnswerResponse,
-  services,
-  translate
-} from '../../../test/util/services.mock';
 import {StoreState} from '../../../reducers';
 import {mapStateToSlidesProps} from '..';
+import {translate} from '../../../../sandbox/translation.mock';
```
Comment (jomaora) [inline packages/@coorpacademy-app-review/src/views/slides/test/on-quit-popin.on-click.test.ts:9]: idem

File: packages/@coorpacademy-app-review/src/views/slides/test/slide.free-text.on-change.test.ts [hunk 15]
```diff
@@ -9,19 +11,14 @@ import {
...
-import {
-  getChoicesCorrection,
-  postAnswerResponses,
-  services,
-  translate
-} from '../../../test/util/services.mock';
 import {createTestStore} from '../../../actions/test/create-test-store';
 import {StoreState} from '../../../reducers';
 import {EDIT_BASIC} from '../../../actions/ui/answers';
+import {translate} from '../../../../sandbox/translation.mock';
```
Comment (jomaora) [inline packages/@coorpacademy-app-review/src/views/slides/test/slide.free-text.on-change.test.ts:17]: idem

File: packages/@coorpacademy-app-review/src/views/slides/test/slide.next-slide.on-click.test.ts [hunk 16]
```diff
@@ -2,23 +2,19 @@ import test from 'ava';
...
-  postAnswerResponses,
-  getChoicesCorrection,
-  incorrectFreeTextPostAnswerResponse,
-  services,
-  translate
-} from '../../../test/util/services.mock';
 import {mapStateToSlidesProps} from '..';
 import {createTestStore} from '../../../actions/test/create-test-store';
 import {NEXT_SLIDE} from '../../../actions/ui/next-slide';
+import {translate} from '../../../../sandbox/translation.mock';
```
Comment (jomaora) [inline packages/@coorpacademy-app-review/src/views/slides/test/slide.next-slide.on-click.test.ts:10]: idem

File: packages/@coorpacademy-app-review/src/views/slides/test/slide.qcm-drag.on-click.test.ts [hunk 17]
```diff
@@ -2,17 +2,19 @@ import test from 'ava';
...
+import {services} from '@coorpacademy/review-services-mocks';
 import {mapStateToSlidesProps} from '..';
-import {ProgressionFromAPI} from '../../../types/common';
-import {services, translate} from '../../../test/util/services.mock';
 import {createTestStore} from '../../../actions/test/create-test-store';
 import {StoreState} from '../../../reducers';
 import {EDIT_QCM_DRAG} from '../../../actions/ui/answers';
 import {QcmDrag} from '../../../types/slides';
-import {qcmDragSlide} from './fixtures/qcm-drag';
+import {translate} from '../../../../sandbox/translation.mock';
```
Comment (jomaora) [inline packages/@coorpacademy-app-review/src/views/slides/test/slide.qcm-drag.on-click.test.ts:12]: idem

File: packages/@coorpacademy-app-review/src/views/slides/test/slide.qcm-graphic.on-click.test.ts [hunk 18]
```diff
@@ -2,17 +2,19 @@ import test from 'ava';
...
+import {services} from '@coorpacademy/review-services-mocks';
 import {mapStateToSlidesProps} from '..';
-import {ProgressionFromAPI} from '../../../types/common';
-import {services, translate} from '../../../test/util/services.mock';
 import {createTestStore} from '../../../actions/test/create-test-store';
 import {StoreState} from '../../../reducers';
 import {EDIT_QCM_GRAPHIC} from '../../../actions/ui/answers';
 import {QcmGraphic} from '../../../types/slides';
-import {qcmGraphicSlide} from './fixtures/qcm-graphic';
+import {translate} from '../../../../sandbox/translation.mock';
```
Comment (jomaora) [inline packages/@coorpacademy-app-review/src/views/slides/test/slide.qcm-graphic.on-click.test.ts:12]: idem

File: packages/@coorpacademy-components/src/organism/review-stacked-slides/style.css [hunk 19]
```diff
@@ -42,35 +40,35 @@
   z-index: 50;
   width: 100%;
   transform: translate(0, 0);
-  transition: transform 0.5s ease-in;
+  transition: transform 1s ease-in, width 1s ease-in;
```
Comment (jomaora) [inline packages/@coorpacademy-components/src/organism/review-stacked-slides/style.css:43]: This changes should not been here in your PR, please check when you build your PR which files have been added. You must be sure that the PR does not includes files changed by other PRs.

File: packages/@coorpacademy-review-services-mocks/src/index.ts [hunk 20]
```diff
@@ -23,12 +23,12 @@ import {freeTextSlide} from './fixtures/free-text';
 import {sliderSlide} from './fixtures/slider';
 import {templateSlide} from './fixtures/template';
 
-const content: ReviewContent = {
+export const content: ReviewContent = {
```
Comment (jomaora) [inline packages/@coorpacademy-review-services-mocks/src/index.ts:26]: not necessary to be exported

File: packages/@coorpacademy-review-services-mocks/src/index.ts [hunk 21]
```diff
@@ -23,12 +23,12 @@ import {freeTextSlide} from './fixtures/free-text';
...
   ref: 'skill_NyxtYFYir',
   type: 'skill'
 };
 
-const engine: ReviewEngine = {
+export const engine: ReviewEngine = {
```
Comment (jomaora) [inline packages/@coorpacademy-review-services-mocks/src/index.ts:31]: same here

File: packages/@coorpacademy-review-services/src/index.ts [hunk 22]
```diff
@@ -8,7 +8,20 @@ import type {
...
+  QcmDragQuestion as _QcmDragQuestion,
+  Question as _Question,
+  QcmGraphicQuestion as _QcmGraphicQuestion,
+  QcmQuestion as _QcmQuestion,
+  SliderQuestion as _SliderQuestion,
+  TemplateQuestion as _TemplateQuestion,
+  ProgressionAnswerItem as _ProgressionAnswerItem,
+  SlideContent as _SlideContent,
+  Rank as _Rank,
+  ChoiceItem as _ChoiceItem
```
Comment (jomaora) [inline packages/@coorpacademy-review-services/src/index.ts:24]: 👍

**Review by jomaora — CHANGES_REQUESTED** [review]

**Review by jomaora — APPROVED** [review]

**Pull request merged**